crates/fluvio-controlplane-metadata/src/topic/config.rs

```diff
@@ -0,0 +1,273 @@
+use std::time::Duration;
+
+use fluvio_types::{ReplicationFactor, TopicName, PartitionCount, IgnoreRackAssignment};
+
+use crate::topic::{
+    ReplicaSpec, TopicReplicaParam, SegmentBasedPolicy, CleanupPolicy, TopicStorageConfig,
+};
+
+use super::{TopicSpec, PartitionMap, CompressionAlgorithm};
+
+const DEFAULT_PARTITION_COUNT: PartitionCount = 1;
+const DEFAULT_REPLICATION_FACTOR: ReplicationFactor = 1;
+const DEFAULT_IGNORE_RACK_ASSIGMENT: IgnoreRackAssignment = false;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "use_serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "kebab-case")
+)]
+pub struct TopicConfig {
```

Comment:
Instead of trying to build manual config that could be brittle and not flexible. better is to use `Builder` strategy similar to ConsumerConfig and other

Reply:
Done